### DISCUSSION TITLE
Chinese / Japanese / Thai text overflow in table cell

### DISCUSSION THREAD

**Discussion by devon254** [dompdf/dompdf#3449] [Q&A] [answered]
Chinese / Japanese / Thai characters overflow in `td` tag of table despite specifying `width: auto`, but content with **ALL** alphanumberic data displays without overflowing issue. 
Seems like chinese characters doesn't word break automatically, when content with newline is in the data before render, there is no overflow. Is there a fix to this or i need to add in additional option configurations?

![firefox_WzGDk98F4n](https://github.com/dompdf/dompdf/assets/141614485/572df0ea-c7fa-4b7a-afe0-481690017d77)

Pompdf version: `2.0.4`
Used font family for chinese characters: `droidsansfallback-cjk`
Options included: `$dompdf->set_paper("A4", "portrait");`

**Comment by bsweeney** [comment]
It does not appear that the characters overflow but that the TD itself is overflowing the page. But it's also hard to say for sure what's going on without a sample of the HTML + CSS. For example, does your table have a defined width? Or if the table is contained inside another element does that element have a defined width?

**Comment by devon254** [comment]
Appreciate your prompt response! I can only get back to you by the next day, but thinking aloud here, why does purely English content doesn't cause this overflow despite them sharing the same width configuration?


________________________________

**Reply by bsweeney** [reply]
One potential cause, which is hard to see here, is that there are no "word" breaks in the text. Dompdf has limited logic for breaking text across lines, primarily limited to typical Latin-based language word breaks (e.g., spaces). I do see spaces in your screen shot, but if the character code isn't for a standard space that could cause problems.

**Reply by devon254** [reply, marked as answer]
Yes, you're right! The lack of word break is causing the overflow and inserting spaces does make it goes to a new line, but there are too many considerations if i were to add in the spaces with a logic code.

I have found that using `table-layout: fixed; width: 100%` on `table` create the word break automatically! But i can't make the `td` of each column to shrink to its longest content but this is already good enough for me. Works well on chinese, japanese and thai characters!


![firefox_sv8TYfiOM1](https://github.com/dompdf/dompdf/assets/141614485/d1d19514-168d-410f-be15-90e718ac3a8f)

**Reply by bsweeney** [reply]
The inability to specify TD width with fixed layout is a known issue, follow #1432 for updates.

You could also try styling the text with `word-break: break-all;` to break between any character. You might have to add a containing DIV on which the styling is applied because of ... other ... issues. You'd need to apply that style only on the CJK text otherwise it would also split words in your latin-based text as well.
